VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowDeployProvider.cs

```diff
@@ -112,17 +114,18 @@ public async Task DeployAsync(CancellationToken cancellationToken, TextWriter te
                 var packageManager = new PackageManager(fileManager);
 
                 outputLogger.Log("Trimming application binaries...");
-
                 await packageManager.TrimApplication(new FileInfo(Path.Combine(outputPath, "App.dll")), osVersion, includePdbs, cancellationToken: cancellationToken);
 
-                await Task.Run(async () => await AppManager.DeployApplication(packageManager, connection, osVersion, outputPath, includePdbs, false, outputLogger, cancellationToken));
```

Comment:
good catch

Reply:
Hmm we might still need something like this. I've just pushed another change to the Extension side of things so that the Deploy happens in the background, but the Progress and Application messages happen on the UI thread. About to test the CI change.